cl/lib/elasticsearch_utils.py

```diff
@@ -667,3 +1050,138 @@ def fetch_es_results(
+    return []
+
+
+def build_has_child_filters(
+    child_type: str, cd: CleanData
+) -> list[QueryString]:
+    """Builds Elasticsearch 'has_child' filters based on the given child type
+    and CleanData.
+
+    :param child_type: The type of child filter to build (e.g., "position").
+    :param cd: The user input CleanedData.
+    :return: A list of QueryString objects containing the 'has_child' filters.
+    """
+
+    queries_list = []
+    if cd["type"] == SEARCH_TYPES.PEOPLE:
+        if child_type == "position":
+            selection_method = cd.get("selection_method", "")
+            court = cd.get("court", "").split()
+            appointer = cd.get("appointer", "")
+            if selection_method:
+                queries_list.extend(
+                    build_term_query(
+                        "selection_method_id",
+                        selection_method,
+                    )
+                )
+            if court:
+                queries_list.extend(build_term_query("court_exact", court))
+            if appointer:
+                queries_list.extend(build_text_filter("appointer", appointer))
+
+    if not queries_list:
+        return []
+
+    return [
+        Q(
+            "has_child",
+            type=child_type,
+            score_mode="max",
+            query=reduce(operator.iand, queries_list),
+            inner_hits={"name": f"filter_inner_{child_type}"},
+            max_children=10,
+        )
+    ]
+
+
+def build_join_es_filters(cd: CleanData) -> List:
+    """Builds join elasticsearch filters based on the CleanData object.
+
+    :param cd: An object containing cleaned user data.
+    :return: The list of Elasticsearch queries built.
+    """
+
+    queries_list = []
+    if cd["type"] == SEARCH_TYPES.PEOPLE:
+        # Build parent document filters.
+        queries_list.extend(
+            build_term_query(
+                "dob_state_id",
+                cd.get("dob_state", ""),
+            )
+        )
+        queries_list.extend(
+            build_term_query(
+                "political_affiliation_id",
+                cd.get("political_affiliation", "").split(),
+            )
+        )
+        queries_list.extend(
+            build_daterange_query(
+                "dob",
+                cd.get("born_before", ""),
+                cd.get("born_after", ""),
+            )
+        )
+        queries_list.extend(
+            build_text_filter(
+                "dob_city",
+                cd.get("dob_city", ""),
+            )
+        )
+        queries_list.extend(
+            build_text_filter(
+                "name",
+                cd.get("name", ""),
+            )
+        )
+        queries_list.extend(
+            build_text_filter(
+                "school",
+                cd.get("school", ""),
+            )
+        )
+
+        # Build position has child filter:
+        queries_list.extend(build_has_child_filters("position", cd))
```

Comment:
There's no a reason to call the extend method six times. I'll refactor this method.

Reply:
We use the extend method six times because each `build_` helper returns a `list`. However, we can accomplish the same result with just one extend method.